pyuvdata/tests/test_uvdata.py

```diff
@@ -1548,3 +1548,366 @@ def test_get_feedpols():
+    ant_pairs_expected = [(1, 3), (1, 6), (2, 3), (2, 6)]
+    pols_expected = [-1, -2, -3, -4]
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_items_equal(polarizations, pols_expected)
+
+    # Antenna numbers with - for avoidance
+    ant_str = '1_(-3,11)'
+    ant_pairs_nums, polarizations = uv.parse_ants(ant_str)
+    ant_pairs_expected = [(1, 11)]
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_is_instance(polarizations, type(None))
+
+    # Remove specific antenna number
+    ant_str = '1,-3'
+    ant_pairs_nums, polarizations = uv.parse_ants(ant_str)
+    ant_pairs_expected = [(0, 1), (1, 2), (1, 6), (1, 7), (1, 8), (1, 11),
+                                      (1, 14), (1, 18), (1, 19), (1, 20), (1, 21),
+                                      (1, 22), (1, 23), (1, 24), (1, 26), (1, 27)]
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_is_instance(polarizations, type(None))
+
+    # Remove specific baseline (same expected antenna pairs as above example)
+    ant_str = '1,-1_3'
+    ant_pairs_nums, polarizations = uv.parse_ants(ant_str)
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_is_instance(polarizations, type(None))
+
+    # Antenna numbers with polarizations and - for avoidance
+    ant_str = '1l_(-3r,11l)'
+    ant_pairs_nums, polarizations = uv.parse_ants(ant_str)
+    ant_pairs_expected = [(1, 11)]
+    pols_expected = [-2]
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_items_equal(polarizations, pols_expected)
+
+    # Antenna numbers and Stokes parameters
+    ant_str = '(1l,2r)_(3l,6r),I,q'
+    ant_pairs_nums, polarizations = uv.parse_ants(ant_str)
+    ant_pairs_expected = [(1, 3), (1, 6), (2, 3), (2, 6)]
+    pols_expected = [2, 1, -1, -2, -3, -4]
+    nt.assert_items_equal(ant_pairs_nums, ant_pairs_expected)
+    nt.assert_items_equal(polarizations, pols_expected)
+
+
+def test_select_with_ant_str():
+    # Test select function with ant_str argument
+    uv = UVData()
+    testfile = os.path.join(
+        DATA_PATH, 'day2_TDEM0003_10s_norx_1src_1spw.uvfits')
+    uvtest.checkWarnings(uv.read_uvfits, [testfile], message='Telescope EVLA is not')
+    inplace = False
+
+    # All baselines
+    ant_str = 'all'
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), uv.get_antpairs())
+    nt.assert_items_equal(uv2.get_pols(),uv.get_pols())
+
+    # Auto correlations
+    ant_str = 'auto'
+    nt.assert_raises(ValueError, uv.select, ant_str=ant_str, inplace=inplace)
+    # No auto correlations in this data
+
+    # Cross correlations
+    ant_str = 'cross'
+    uv2 = uv.select(ant_str=ant_str, inplace=inplace)
+    nt.assert_items_equal(uv2.get_antpairs(), uv.get_antpairs())
+    nt.assert_items_equal(uv2.get_pols(), uv.get_pols())
+    # All baselines in data are cross correlations
```

Comment:
You also need a test with a data set that has auto correlations so you can test that this doesn't just always return everything.